regolith/helpers/u_milestonehelper.py

```diff
@@ -132,16 +125,23 @@ def db_updater(self):
         if rc.index > 1:
             doc = all_milestones[rc.index-2]
             identifier = doc['identifier']
-            if not rc.type and not doc.get('type'):
+            if rc.type not in (list(chain.from_iterable((k, v) for k, v in ALLOWED_TYPES.items()))):
+                if not doc.get('type'):
                     raise RuntimeError(f"please rerun specifying --type with a value from {ALLOWED_TYPES}")
             if rc.type:
                 try:
+                    for k, v in ALLOWED_TYPES.items():
```

Comment:
now, if we still need this, it should be moved down to where the type is assigned to the update so the logic is clearer to the reader, unless there is some reason in can't be.

Also, shouldn't need the try except because we have already validated.

Also, don't we need the same treatment for stati (plural of status)

Reply:
@sbillinge does it looks better right now? 
I am checking at the beginning if the type and status is valid!